src/EventSourcing/DomainMessageBuilder.php

```diff
@@ -40,6 +40,13 @@ class DomainMessageBuilder
      */
     private $forReplay;
 
+    private UuidFactoryInterface $uuidFactory;
+
+    public function __construct(UuidFactoryInterface $uuidFactory)
```

Comment:
You could work with a default parameter here to avoid lots of changes
```
    public function __construct(UuidFactoryInterface $uuidFactory = null)
    {
        if ($uuidFactory === null) {
           $uuidFactory = new UuidFactory(); 
        }
        
        $this->uuidFactory = $uuidFactory;
    }
```

Reply:
Normally I would say to avoid using default parameters, but this is existing code and it was previously self-contained. Now the extra parameter gives some overhead.